sdk/servicebus/azure_messaging_servicebus/src/client.rs

```diff
@@ -0,0 +1,552 @@
+        _options: Option<CreateSenderOptions>,
+    ) -> Result<Sender> {
+        // Authorize the path if we have a credential-based client
+        self.authorize_path(queue_or_topic_name).await?;
+
+        Sender::new(
+            self.connection.clone(),
+            queue_or_topic_name.to_string(),
+            self.options.clone(),
+        )
+        .await
+    }
+
+    /// Creates a receiver for the specified queue with options.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use azure_identity::DeveloperToolsCredential;
+    /// use azure_messaging_servicebus::{ServiceBusClient, CreateReceiverOptions, ReceiveMode, SubQueue};
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    ///     // ... create client ...
+    ///     let credential = DeveloperToolsCredential::new(None)?;
+    ///     let client = ServiceBusClient::builder()
+    ///         .open("my-servicebus.servicebus.windows.net", credential.clone()).await?;
+    ///
+    ///     // Create receiver with default PeekLock mode
+    ///     let receiver = client.create_receiver("my-queue", None).await?;
+    ///
+    ///     // Create receiver with ReceiveAndDelete mode
+    ///     let receiver = client.create_receiver("my-queue", Some(CreateReceiverOptions {
+    ///         receive_mode: ReceiveMode::ReceiveAndDelete,
+    ///         sub_queue: None,
+    ///     })).await?;
+    ///
+    ///     // Create receiver for dead letter queue
+    ///     let receiver = client.create_receiver("my-queue", Some(CreateReceiverOptions {
+    ///         receive_mode: ReceiveMode::PeekLock,
+    ///         sub_queue: Some(SubQueue::DeadLetter),
+    ///     })).await?;
+    ///     Ok(())
+    /// }
+    /// ```
+    pub async fn create_receiver(
+        &self,
+        queue_name: &str,
+        options: Option<CreateReceiverOptions>,
+    ) -> Result<Receiver> {
+        // Build the entity path based on sub_queue option
+        let options = options.unwrap_or_default();
+        let entity_path = if let Some(ref sub_queue) = options.sub_queue {
+            format!("{}{}", queue_name, sub_queue.as_path_suffix())
+        } else {
+            queue_name.to_string()
+        };
+
+        // Authorize the path if we have a credential-based client
+        self.authorize_path(&entity_path).await?;
+
+        Receiver::new(
+            self.connection.clone(),
+            entity_path,
+            None,
+            options.receive_mode,
+            self.options.clone(),
+        )
+        .await
+    }
+
+    /// Creates a receiver for the specified topic and subscription with options.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use azure_identity::DeveloperToolsCredential;
+    /// use azure_messaging_servicebus::{ServiceBusClient, CreateReceiverOptions, SubQueue};
+    ///
+    /// #[tokio::main]
+    /// async fn main() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Examples don't typically show boiler plate. Hide like we do in other methods. Preface these with a `#` and unindent the following lines. Don't forget to hide the final `Ok(())` and `}` with a `#` as well. That's not only idiomatic, it's consistent throughout our codebase (or should be).